feat(aio): initial design implementation

sjtrimble opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe:
Design

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

gkalpak commented on aio/src/styles/_typography.scss:
```diff
@@ -1,26 +1,39 @@
+      font-size: 24px;
+      font-weight: 500;
+      margin: 12px 0px;
+      text-transform: uppercase;
```
This is breaking some tests :)

gkalpak replied:
Changing [this line] as shown below should fix the tests:

```diff
-expect(page.getDocViewerText()).toContain('Progressive web apps');
+expect(page.getDocViewerText()).toMatch(/Progressive web apps/i);
```

[1]: https://github.com/sjtrimble/angular/blob/3762ebe1b16a6a5d01cffb31e87c46b4b42cc6f8/aio/e2e/app.e2e-spec.ts#L13

petebacondarwin commented:
I think we need to add `.sass-cache` to the `.gitignore`

petebacondarwin commented on aio/src/styles/_animations.scss:
```diff
@@ -0,0 +1,9 @@
+}
+
+button.hambuger.mat-icon:hover {
+    color: red;
```
should this be using a Sass variable, like `$brightred`?

sjtrimble replied:
It will likely be the $offwhite, but this was more a reminder to me to change later. Should I comment out for now?

petebacondarwin commented:
@sjtrimble - I think you ran `yarn install` in the root folder of the project, which created that `yarn.lock` file. We don't use yarn in the root yet, only inside the `aio` folder. So that file needs removing.

alfaproject commented:
@petebacondarwin or use node-sass instead of Compass (:

sjtrimble commented:
@gkalpak - could you take a look at this Travis test that is failing? This one is new since the last few PRs got approved and wasn't happening to my build earlier.

IgorMinar commented on aio/src/app/layout/top-menu/top-menu.component.ts:
```diff
@@ -3,7 +3,7 @@ import { NavigationNode } from 'app/navigation/navigation.service';
 @Component({
   selector: 'aio-top-menu',
-  template: `<span *ngFor="let node of nodes"><a class="nav-link" [href]="node.path || node.url">{{ node.title }}</a></span>`,
+  template: `<li *ngFor="let node of nodes"><a class="nav-link" [href]="node.path || node.url">{{ node.title }}</a></li>`,
```
is this change correct? I believe that `li` can be present only within an `ul` container.

IgorMinar commented on aio/src/index.html:
```diff
@@ -21,7 +21,8 @@
 
   <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
-
+<link href="https://fonts.googleapis.com/css?family=Lato:400,700" rel="stylesheet">  <link href="https://fonts.googleapis.com/css?family=Droid+Sans+Mono" rel="stylesheet">
```
please indent with 2 spaces

IgorMinar commented on aio/src/styles/_animations.scss:
```diff
@@ -0,0 +1,9 @@
+@import 'constants';
+
+// .hamburger.mat-button {
```
delete dead code :)

petebacondarwin commented:
@sjtrimble the Travis failure is a flake that should be fixed by #15049

petebacondarwin commented on .gitignore:
```diff
@@ -30,3 +30,4 @@ yarn-error.log
 # angular.io
 /aio/src/content/docs
 /aio/dist
+/aio/sass-cache
```
And also remove the cached files from this PR :-P

sjtrimble replied:
So we want line 33 removed now? I thought we had said we wanted these ignored? Or are you saying something else?

petebacondarwin replied:
No, keep that line. But your PR includes two files that ought to be ignored (by this setting).
So you will need to actively remove them from the commits

gkalpak replied:
No, line 33 should stay (although probably modified to `/aio/.sass-cache`).
You should remove cached files that have been accidentally commit (before including this rule to make git ignore them). E.g. aio/.sass-cache/e42f7e0f131840ee68ba5e34f73a6a2903ca2b2b/main.scssc

sjtrimble replied:
I removed the cached files/folder from the commit as well 👍 thanks for clarifying.

gkalpak replied:
Does this needs to be `/aio/sass-cache` (without a dot) or `/aio/.sass-cache` (with a dot)?

petebacondarwin replied:
with a dot I think

petebacondarwin commented:
You can remove the text "Home" from the top nav by removing these lines here: https://github.com/angular/angular/blob/master/aio/src/content/navigation.json#L3-L6

petebacondarwin commented:
This is starting to look really nice @sjtrimble. I like the typography in the body.

A couple of layout issues:

1. The first item in the top bar is always too low.
![screen shot 2017-03-13 at 06 44 13](https://cloud.githubusercontent.com/assets/15655/23844196/8cc620d4-07b8-11e7-969f-53df8b4c0534.png)

2. The padding on the search box seems out.
![screen shot 2017-03-13 at 06 41 58](https://cloud.githubusercontent.com/assets/15655/23844205/a4112acc-07b8-11e7-98b5-5c2e269f238e.png)

3. In narrow mode the topbar moves down below the hamburger but it is not being displayed correctly.
![screen shot 2017-03-13 at 06 48 50](https://cloud.githubusercontent.com/assets/15655/23844323/4713089e-07b9-11e7-92de-4a9e5f000225.png)

4. I am not keen on the horizontal rule below the main heading. It seems unnecessary to me.
![screen shot 2017-03-13 at 06 50 55](https://cloud.githubusercontent.com/assets/15655/23844356/822868f2-07b9-11e7-8fe4-01de12edbe1c.png)

petebacondarwin commented on aio/src/app/app.component.html:
```diff
@@ -1,12 +1,18 @@
 <md-toolbar color="primary" class="app-toolbar">
   <button *ngIf="isHamburgerVisible" class="hamburger" md-button (click)="sidenav.toggle()"><md-icon>menu</md-icon></button>
-  <aio-top-menu *ngIf="isSideBySide" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu>
+  <aio-top-menu role="navigation" *ngIf="isSideBySide" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu>
```
can we move this to the `ul` in the template for `aio-top-menu`?
Otherwise we need to add it to the other "small" `aio-top-menu` below.

sjtrimble replied:
That should make sense I think - I was trying to address ul/li items and accessibility a little bit with this.

petebacondarwin commented on aio/src/app/layout/top-menu/top-menu.component.ts:
```diff
@@ -4,8 +4,10 @@ import { NavigationNode } from 'app/navigation/navigation.service';
   template: `
-    <span><a class="nav-link home" href="/"><img src="{{ homeImageUrl }}" title="Home" alt="Home"></a></span>
-    <span *ngFor="let node of nodes"><a class="nav-link" [href]="node.path || node.url">{{ node.title }}</a></span>`,
+    <ul>
```
put the `role="navigation"` attribute here

sjtrimble replied:
That should make sense I think - I was trying to address ul/li items and accessibility a little bit with this.

petebacondarwin commented on aio/src/styles/main.scss:
```diff
@@ -1,31 +1,43 @@
 
 $theme: mat-light-theme($primary, $accent);
 @include angular-material-theme($theme);
+*/
```
why not just delete all this stuff?

sjtrimble replied:
We sure can :) Initially I didn't want to start deleting things that others may have been working on, but we shouldn't need this as it's addressed in the ng-io-theme.scss file.

petebacondarwin reviewed (changes requested):
The visuals look better. I understand that the search box and narrow width topbar will be fixed in an second PR.

I made a couple of minor comments, which should be quick to fix.

I am afraid that this will need considerable rebasing when the search PRs (#15040 and #15096) land, which hopefully will be pretty soon.

I think it is probably worth waiting for them to merge, or if you are feeling brave rebasing on #15096.

petebacondarwin reviewed (approved):
LGTM - I have added items to our TODO list for the search box and mini-toobar

sjtrimble committed:
feat(aio): initial design commit

Implemented style guide elements to top header bar, side hamburger nav menu, content and search. Consolidated SCSS files to styles folder. Fixed PWA test.

gkalpak commented on aio/src/app/app.component.ts:
```diff
@@ -10,11 +10,10 @@ import { SearchService } from 'app/search/search.service';
 @Component({
   selector: 'aio-shell',
   templateUrl: './app.component.html',
-  styleUrls:  ['./app.component.scss'],
```
The file (`app.component.scss`) seems to be still there. If it is not needed any more, then remove it.

sjtrimble replied:
Done.

gkalpak commented on aio/src/index.html:
```diff
@@ -32,13 +32,14 @@
   <!-- End Google Analytics -->
 
+  <link href="https://fonts.googleapis.com/css?family=Lato:400,700" rel="stylesheet">
+  <link href="https://fonts.googleapis.com/css?family=Droid+Sans+Mono" rel="stylesheet">
```
Let's move these next to the other `fonts.googleapis.com` above (on line 23).

sjtrimble replied:
Moved this up along with the others. Too much refactor! :P

gkalpak commented on aio/src/styles/1-layouts/_content-layout.scss:
```diff
@@ -0,0 +1,24 @@
+.sidenav-content {
+    padding: 1rem 3rem;
+    height: 100vh;
+    margin: auto;
```
Inconsistent indentation 😱

sjtrimble replied:
Fixed!

gkalpak commented on aio/src/styles/1-layouts/_sidenav.scss:
```diff
@@ -42,17 +42,23 @@
+    box-shadow: 6px 0 6px rgba(0,0,0,0.10);
+    background-color: $offwhite;
+    padding: 10px 0px 0px;
+    min-width: 220px;
```
Inconsistent indentation 😱

sjtrimble replied:
And, also fixed!

sjtrimble committed:
refactor(aio): indent and org cleanup

sjtrimble commented on aio/src/app/app.component.html:
```diff
@@ -1,11 +1,11 @@
 </md-toolbar>
 
-<md-sidenav-container class="sidenav-container">
+<md-sidenav-container class="sidenav-container" (window:resize)="onResize($event.target.innerWidth)">
```
I removed this.

IgorMinar commented:
@gkalpak how come there are no previews for this PR? is it because @sjtrimble hasn't pushed anything to this branch since we merged the aio-master commit into master or is it because we haven't yet added aio-contributors to the ngbuils.io server setup?

sjtrimble committed:
feat(aio): top bar typography, sidenav size

gkalpak commented:
@igorminar, afaict it is because we need to merge the commit updates `deploy.sh` into `aio-master` (last time I checked it was only in `master`).

gkalpak commented:
@igorminar, actually now it is because I haven't reloaded the upload server to take the new value for teams into account :(

mary-poppins commented:
The angular.io preview for bd00846 is available [here][1].

[1]: https://pr15046-bd00846989d1b9dcd6ad5af23fcccc78dba2b316.ngbuilds.io/
